ipi_tests/examples/exampletools.py

```diff
@@ -183,7 +183,17 @@ def run(self, cwd, nid):
                 stdout=sp.PIPE,
                 stderr=sp.PIPE,
             )
-            time.sleep(3)
+
+            if len(clients) > 0:
+                f_connected = False
+                for i in range(50):
+                    if os.path.exists("/tmp/ipi_" + clients[0][1]):
+                        f_connected = True
+                        break
+                    else:
+                        time.sleep(0.1)
+                if not f_connected:
+                    raise RuntimeError("Couldn't find the i-PI UNIX socket")
```

Comment:
The clients are called afterward,
I don't get what are these lines for

Reply:
This substitutes the `sleep(3)`.  It waits until i-PI has created the sockets before letting the clients start.